server-spi-private/src/main/java/org/keycloak/requestfilter/AccessRules.java

```diff
@@ -0,0 +1,166 @@
+package org.keycloak.requestfilter;
+
+import io.netty.handler.ipfilter.IpFilterRuleType;
+import io.netty.handler.ipfilter.IpSubnetFilterRule;
+import org.jboss.logging.Logger;
+import org.keycloak.utils.StringUtil;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+class AccessRules {
+
+    private final List<AccessRule> rules;
+
+    private AccessRules(List<AccessRule> rules) {
+        this.rules = rules;
+    }
+
+    public boolean matches(InetSocketAddress address, String requestPath) {
+
+        for (AccessRule accessRule : rules) {
+            if (accessRule.matches(address, requestPath)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    static AccessRules parse(String rulesString, String contextPath, IpFilterRuleType ruleType) {
+        return (new AccessRulesParser(rulesString, contextPath, ruleType)).parse();
+    }
+
+    static class IpRuleConfigurationException extends RuntimeException {
+        IpRuleConfigurationException(String msg) {
+            super(msg);
+        }
+    }
+
+    private static class AccessRulesParser {
+
+        private static final Logger log = Logger.getLogger(AccessRulesParser.class);
+        private final String rulesString;
+        private final String contextPath;
+        private final IpFilterRuleType ruleType;
+
+        private final List<String> cidrs = new ArrayList<>();
+
+        private AccessRulesParser(String rulesString, String contextPath, IpFilterRuleType ruleType) {
+            this.rulesString = rulesString;
+            this.contextPath = contextPath;
+            this.ruleType = ruleType;
+        }
+
+        private AccessRules parse() {
+            if ((this.rulesString == null) || StringUtil.isBlank(this.rulesString)) {
+                log.info("No access rules");
+                return null;
```

Comment:
Instead of `null` you could return an `EMPTY`  null-object  `AccessRules` instance.